File: 프로그래머스/unrated/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

class Solution {
    public String[] solution(String myString) {
        String[] arr = myString.split("x");
        // myString을 x를 기준으로 배열로 변환한다
		Arrays.sort(arr);
        // 배열을 정렬한다
		List<String> list = new ArrayList<String>();
		
		for (int i = 0; i < arr.length; i++) {
		    String str = arr[i];
		    if (!str.isEmpty()) {
		        list.add(str);
		    }
	    } // 배열의 원소가 비어있지 않다면 list에 추가한다

		String answer[] = list.toArray(new String[list.size()]);
        // list를 배열로 재 변환한다
        return answer;
    }
}

File: 프로그래머스/0/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.js
function solution(myString) {
    const arr = myString.split('x');
    // 문자열을 'x'를 기준으로 배열로 분할
    
    arr.sort();
    // 배열을 정렬
    
    const list = arr.filter(str => str !== '');
    // 빈 문자열이 아닌 원소만 새로운 배열에 추가
    return list;
}
